### DISCUSSION TITLE
ansible-lint line length exceeded and ssh pub keys

### DISCUSSION THREAD

**Discussion by RobVerduijn** [ansible/ansible-lint#3223] [Q&A]
Hi,

I noticed an interesting thing about the ansible-lint line length exceeded error.
It's fine when you're doing regular stuff, and most issues can be avoided by using [yaml multi line tricks](https://yaml-multiline.info/)

But when you set up a host_var in host_vars/my_host.yml using the special ansible variable ansible_pubkey the line length exceeds the 160 character limit quite a bit

576 chars is the line length in my case ( pub key generated with ssh-keygen/openssh-8.8p1 on fedora37)
So snipping of my email and using multiline '>-' isn't really putting a dent in that number. 
(there are 2 spaces in the entire line, the one after the ssh-rsa in the beginning and the one before the email at the end)

And disabling the length exceeded check globally isn't really what I'm looking for.

Any ideas on how to deal with this ?

Rob

**Comment by AndrewKvalheim** [comment]
[Double-quoted flow scalars](https://yaml.org/spec/1.2.2/#731-double-quoted-style):

```yaml
example: "ssh-rsa xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
  xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
  xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
  xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
  xxxxxxxxxxxxxxxxxxxx example"
```

**Reply by muellert** [reply]
That might be fine as long as you are "inside" the scope of YAML, but breaks down if you use the YAML multiline facility to embed a snippet of a configuration file that is not YAML. Eg. I am using this role: geerlingguy.nginx, where, for many purposes, you need to embed a piece of nginx configuration into a variable. Amongst those, I have a specification for SSL cipers:

```
extra_parameters: |
    ...
    ssl_ciphers          "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RS..." (long line)
    ...
```

ansible-lint detects the colon and then complains about weird stuff, plus about the line length. But since this isn't YAML, I can't just use YAML's multiline facilities. Working around this is also a kludge, as the snippet is somewhere inside my inventory. I haven't yet found a '#noqa' incantation to silence the error message from ansible-lint.

**Reply by RobVerduijn** [reply]
> [Double-quoted flow scalars](https://yaml.org/spec/1.2.2/#731-double-quoted-style):
> 
> ```yaml
> example: "ssh-rsa xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
>   xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
>   xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
>   xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
>   xxxxxxxxxxxxxxxxxxxx example"
> ```

It looks to me like you neatly escaped the linebreaks with \\, but sadly inserted 2 spaces on every line into the pubkey.
Even if this is not happening, this method seems very prone to human error.

Ansible-lint is a tool that helps you write neat ansible code, this seems more like a solution to make ansible code support the tool.

right now the only real workable solution is to add the skip ansible lint tag
maybe even better using the new skip ansible lint rule tag https://github.com/ansible/ansible-lint/issues/364

**Reply by AndrewKvalheim** [reply]
You asked for ideas on how to deal with this. Your options are to either disable the lint, propose new behavior of the linter, or write shorter lines. You described having difficulty figuring out how to write shorter lines.

> It looks to me like you neatly escaped the linebreaks with \\, but sadly inserted 2 spaces on every line into the pubkey.

This is trivial for you to test.

**Reply by muellert** [reply]
@AndrewKvalheim I'm not sure what the question is. I thought we're discussing long lines here, and while I understand that there are ways to work around this in pure Yaml, I'm lacking a way to disable this check for code that is technically within a playbook, but can't adhere to Yaml conventions because it has different syntax and semantics (nginx configuration file, in my case). It would be nice to have a way to tell ansible-lint to ignore such embedded blocks.

**Reply by AndrewKvalheim** [reply]
@muellert I have no idea what question you’re talking about. I have been ignoring you because it sounds like you’re describing a different problem and I don’t understand why you’re raising it here.

**Reply by mickaelperrin** [reply]
> [Double-quoted flow scalars](https://yaml.org/spec/1.2.2/#731-double-quoted-style):
> 
> ```yaml
> example: "ssh-rsa xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
>   xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
>   xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
>   xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\
>   xxxxxxxxxxxxxxxxxxxx example"
> ```

That's interesting, I am struggling on how to use ansible-lint with my repo. I have some pub keys in ansible variables, and can't get a way to pass the linter fix/check.

Currently, the double-quoted flow scalars syntax is automatically transformed by the ansible-lint --fix command to a single line that exceeds the line length.

As ansible-lint, check/fix the syntax of every files in the repo, I am unable to have a commit working with the precommit hook using the ansible linter. I have to use the "--no-verify" argument to every git commit currently.

I don't want to disable the rule because it worked well for every other types of lines where I can have spaces.

Any hint ?

**Reply by AndrewKvalheim** [reply]
Sounds like you’re describing #4233

**Reply by mickaelperrin** [reply]
Interesting, but I don't have a custom .yamllint file.

See the following file and repo, I reproduced the issue in a minimal example:  https://github.com/mickaelperrin/ansible-lint-issues/blob/master/roles/ansible_lint_tests/tasks/main.yml
